libpod/runtime_img.go

```diff
@@ -276,24 +277,51 @@ func DownloadFromFile(reader *os.File) (string, error) {
 
 // LoadImage loads a container image into local storage
 func (r *Runtime) LoadImage(ctx context.Context, inputFile string, writer io.Writer, signaturePolicy string) (string, error) {
-	if newImages, err := r.LoadAllImageFromArchive(ctx, writer, inputFile, signaturePolicy); err == nil {
+	newImages, err := r.LoadAllImageFromArchive(ctx, writer, inputFile, signaturePolicy)
+	if err == nil {
 		return newImages, nil
 	}
+	logrus.Warnf("Error trying pulling multi-arch images: %v", err)
 	return r.LoadImageFromSingleImageArchive(ctx, writer, inputFile, signaturePolicy)
 }
 
 // LoadAllImageFromArchive loads all images from the archive of multi-image that inputFile points to.
 func (r *Runtime) LoadAllImageFromArchive(ctx context.Context, writer io.Writer, inputFile, signaturePolicy string) (string, error) {
-	newImages, err := r.ImageRuntime().LoadAllImagesFromDockerArchive(ctx, inputFile, signaturePolicy, writer)
+	var latestErr error
+	appendErr := func(e error) {
+		if latestErr != nil {
+			latestErr = errors.Wrapf(latestErr, "%v\n", e)
+		} else {
+			latestErr = e
+		}
+	}
+	newImageNames, err := r.ImageRuntime().LoadAllImagesFromDockerArchive(ctx, inputFile, signaturePolicy, writer)
+	if err == nil {
+		return strings.Join(newImageNames, ", "), nil
+	}
+	appendErr(errors.Wrapf(err, "error trying multi docker aechive"))
```

Comment:
“archive”, and this probably could benefit from a bit of rewording (users don’t care much about the “multi” part) — I haven’t tried this to see what the eventual full error text looks like.

(In both cases).